contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -33,23 +33,16 @@ fi
 
 # if pool encryption is active and the zfs command understands '-o encryption'
 if [[ $(zpool list -H -o feature@encryption $(echo "${root}" | awk -F\/ '{print $1}')) == 'active' ]]; then
-    # check if root dataset has encryption enabled
-    if $(zfs list -H -o encryption "${root}" | grep -q -v off); then
-        # figure out where the root dataset has its key, the keylocation should not be none
-        while true; do
-            if [[ $(zfs list -H -o keylocation "${root}") == 'none' ]]; then
-                root=$(echo -n "${root}" | awk 'BEGIN{FS=OFS="/"}{NF--; print}')
-                [[ "${root}" == '' ]] && exit 1
-            else
-                break
-            fi
-        done
+    # if the root dataset has encryption enabled
+    ENCRYPTIONROOT=$(zfs get -H -o value encryptionroot ${ZFS_DATASET})
+    if [[ $(echo "${ENCRYPTIONROOT}" | grep -v -E '^$|^-$') ]]; then
```

Comment:
Use `[ "${ENCRYPTIONROOT}" == "-" ]` instead?

Reply:
yeah, wasn't able to test this, but I'll make that change too. thanks for the reminder.